libc/misc/popen.c

```diff
@@ -5,37 +5,36 @@
 
 FILE * popen(char *command, char *rw)
 {
-   int pipe_fd[2];
-   int pid, reading;
+    int pipe_fd[2];
+    int pid, reading;
 
-   if( command == NULL || pipe(pipe_fd) < 0 ) return NULL;
-   reading = (rw[0] == 'r');
+    if( command == NULL || pipe(pipe_fd) < 0 ) return NULL;
+    reading = (rw[0] == 'r');
 
-   pid = vfork();
-   if( pid < 0 ) { close(pipe_fd[0]); close(pipe_fd[1]); return NULL; }
-   if( pid == 0 )
-   {
-      close(pipe_fd[!reading]);
-      close(reading);
-      if( pipe_fd[reading] != reading )
-      {
-	 dup2(pipe_fd[reading], reading);
-         close(pipe_fd[reading]);
-      }
+    pid = vfork();
+    if( pid < 0 ) { close(pipe_fd[0]); close(pipe_fd[1]); return NULL; }
```

Comment:
@ccoffing, when you re-tab functions for readability, what do you think of the idea of also reformatting them to an agreed-on ELKS standard? In this case, the above line in this function uses poor style, IMO. Do you have an indentation program that can be used when retabbing files that have ancient style? I pushed `elks/scripts/indent.sh` which can be used, I suggest that for this file since we're retabbing it entirely anyways. 

The same could also possibly be done for the other files touched for fixups, for instance libc/termcap/tputs.c whose formatting also seems ancient. In that file, it might be a good idea to update function argument decls to ANSI rather than K&R.

Reply:
I like using tools to format.  It's not perfect, but it's certainly nice to have the options checked into the source tree both to drive the tool and as documentation of what the standard even is.

That said, I first improved the formatting by hand (that's what's in the PR just merged) then ran indent using the options in the script (for my education).  The `indent` program on Debian Linux doesn't understand the `-ldi8` option.  But my FreeBSD box does understand it.  And the output differs slightly between the two (besides just due to `ldi`).  Interesting.

Reply:
> The indent program on Debian Linux doesn't understand the -ldi8 option. But my FreeBSD box does understand it. And the output differs slightly between the two 

Geez, I didn't even think of the problem of multiple versions of indent acting differently with the same command line. I suppose we also need to describe the desired output format somewhere as well, as you mention. (The file `coding-style.txt` in the root directory is out of date, directly from a very old version of the Linux kernel over 20 years ago).

> besides just due to ldi

Actually, the local variable formatting is my least favorite part of the indent.sh script using -ldi8, as it indents the variable names after the type(s), which I'm not sure is great. But on my macOS system, I had to do something, or indent would output everything on column 0!

Do we need another version/option for Debian Linux? I'm open to other ideas, I just happened to have an indent spec from some of my other projects handy so I used that. I'm not that familiar with all the tools but I understand there may be some newer tools that are superior.

Reply:
Regarding `-di8 -ldi8`, if I understand what you're saying, maybe what you want instead is `-di1` (and that looks better to me too).

I realized that the different behavior I was seeing, was due to GNU indent switching the defaults to their own style.  Since your indent.sh does not specify every setting, I was getting a mis-mash of settings.  And worse, GNU indent only has some shared settings with the original indent.  Forget I even confused the issue with GNU indent.

I suggest creating a `indent.pro` in the root directory of the project, with all settings explicitly listed (including, probably, `-di1`).  Then developers can copy it to `~/.indent.pro` or explicitly use `indent -Pindent.pro`.  And format from Mac or a BSD, or original indent.c built on Linux (which is what I did).  Works fine.